app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -4,60 +4,60 @@ module VendorConfigurations
       # Mock JWTConfiguration class to use it to develop other part in the jwt authenticator
       class ConfigurationJWTGenericVendor < ConfigurationInterface
 
-        def initialize
+        def initialize(authenticator_input)
           @authentication_parameters_class = Authentication::AuthnJwt::AuthenticationParameters
           @extract_token_from_credentials = Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
           @validate_and_decode_token_class = Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
           @identity_provider_factory = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
           @validate_resource_restrictions_class = Authentication::ResourceRestrictions::ValidateResourceRestrictions
           @create_identity_provider_class = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
           @logger = Rails.logger
+          authentication_parameters(authenticator_input)
         end
 
-        def authentication_parameters(authenticator_input)
-          @logger.debug(LogMessages::Authentication::AuthnJwt::CREATING_AUTHENTICATION_PARAMETERS_OBJECT.new)
-          jwt_token = @extract_token_from_credentials.call(
-            credentials: authenticator_input.request.body.read
-          )
-          @authentication_parameters_class.new(
-            authentication_input: authenticator_input,
-            jwt_token: jwt_token
-          )
+        def jwt_identity
+          @jwt_identity ||= jwt_identity_from_request
         end
 
-        def jwt_identity(authentication_parameters)
-          identity_provider = @create_identity_provider_class.new.call(
-            authentication_parameters: authentication_parameters
-          )
-          identity_provider.provide_jwt_identity
-        end
-
-        def validate_restrictions(authentication_parameters)
+        def validate_restrictions
           initialize_validate_restrictions
           @validate_resource_restrictions.call(
```

Comment:
instead of calling the method `initialize_validate_restrictions` and then calling `@validate_resource_restrictions`, how about creating a new class called `ValidateJwtResourceRestrictions`? this class can have the object that are initialized in `initialize_validate_restrictions` in its dependencies and it will call `@validate_resource_restrictions`.

wdyt?

Reply:
out of scope of this pr for authentication parameters. Can you comment this on the pr for the master?

Reply:
done